# Практическая работа №3: Обработка выборочных данных. Нахождение интервальных оценок параметров распределения. Проверка статистической гипотезы о нормальном распределении

Выполнили студенты гр. 2381 Комосский Егор Артемович и Рыжиков Иван Сергеевич. Вариант №17.

## Цель работы

Получение практических навыков вычисления интервальных статистических оценок параметров распределения выборочных данных и проверки «справедливости» статистических гипотез.

## Основные теоретические положения

**Интервальная оценка** — статистическая оценка параметра $\theta$, которая задаётся двумя числами($(\theta^* − \delta; \theta^* + \delta)$), где $\delta$ — точность, $\gamma = P{|\theta^* − \theta| < \delta}$ — надёжность (доверительная вероятность). 

**Уровень значимости** - $q = 1 − \gamma$.

**Доверительный интервал для математического ожидания при неизвестном СКО**:

$$
\left( \bar x_в - \frac{t_\gamma S}{\sqrt{N}}, \bar x_в + \frac{t_\gamma S}{\sqrt{N}} \right)
$$, 
где $t_\gamma$ — коэффициент Стьюдента.

**Доверительный интервал для среднеквадратичного отклонения**:

$$
(S(1 - q), S(1 + q))
$$, 
где $q$ — табличный коэффициент, зависящий от объема выборки $N$ и надежности $\gamma$.

**Статистической гипотезой** называют гипотезу о виде неизвестного распледеления исследуемой случайной величиный или о значении параметров известных распределений.

Выдвинутую гипотезу называют **нулевой или основной гипотезой** и называют $H_0$.

**Конкурирующей или альтернативной** называют гипотезу $H_1$.

**Односторонний критерий** — это критерий проверки статистической гипотезы, при котором альтернативная гипотеза задаёт отклонение параметра только в одну сторону:
$$H_0: \theta \geq \theta_0, \qquad H_1: \theta < \theta_0 $$
или
$$
H_0: \theta \leq \theta_0, \qquad H_1: \theta > \theta_0.
$$

**Двухсторонний критерий** — это критерий проверки статистической гипотезы, при котором альтернативная гипотеза задаёт отклонение параметра в обе стороны:
$$
H_0: \theta = \theta_0, \qquad H_1: \theta \neq \theta_0.
$$о 
**Ошибка I рода** — отвергается верная $H_0$, хотя она верна.

**Ошибка II рода** — принимается неверная $H_0$, хотя она неверна.

**Мощность критерия** - вероятность совершения ошибки первого рода, обознают как $\alpha$.

**Критерий Пирсона**:

$$
\chi^2_{набл} = \sum_{i=1}^{K} \frac{(n_i - n_i')^2}{n_i'}
$$,
где $n_i'$ — теоретические частоты — частоты, с которыми нормально распределенная величина с параметрами $\bar{x}_в$ и $S$ попадала бы в $i$-й интервал интервального ряда при проведении $N$ испытаний.

Теоретические частоты вычисляются по формуле:

$$
n_i' = N \cdot p_i
$$, 

где $i = 1, 2, 3, ..., K$

$p_i$(вероятность попадания случайной величины в каждый $i$-й интервал интервального ряда) вычисляется по формуле:

$$
p_i = \Phi (z_{i+1}) - \Phi (z_i)
$$,
где $i = 1, 2, 3, ..., K$; 

$\Phi$ - функция Лапласа:

$$
\Phi (z) = \frac{1}{\sqrt{2\pi}} \int_0^z e^{-\frac{t^2}{2}} dt
$$

$z_i$(значения границ интервалов интервального ряда) вычисляется по формуле:
$$
z_i = \frac{x_i - \bar x_в}{S}
$$, 
где $i = 2, 3, ..., K$

**Областью принятия нулевой гипотезы** называют совокупность значения критерия, при которых нулевую гипотезу принимают.

**Критической областью** называют совокупность значения критерия, при которых нулевую гипотезу отвергают.

**Гипотеза о нормальном распределении** принимается, если выполняется:

$$
\chi^2_{набл} \leq \chi^2_{крит}
$$

**Число степеней свободы:**

$$
k = K - 3
$$исло оценённых параметров нормального распределения ($\mu$ и $\sigma$).

## Постановка задачи

Для заданной надежности определить (на основании выборочных данных и результатов выполнения практической работы №2) границы доверительных интервалов для математического ожидания и среднеквадратичного отклонения случайной величины. Проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия Пирсона $\chi^2$. Дать содержательную интерпретацию полученным результатам.

## Порядок выполнения работы

1. Вычислить точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении при заданном объёме выборки для доверительной точности $\gamma \in \{0.95, 0.99\}$. Сделать выводы.
2. Для вычисления границ доверительного интервала для среднеквадратичного отклонения определить значение $q$ при заданных $\gamma$ и $n$. Построить доверительные интервалы, сделать выводы.
3. проверить гипотезу о нормальном распределении исследуемой случайной величины с помощью критерия $\chi^2$(Пирсона). Для этого необходимо найти теоретические частоты и вычислить наблюдаемое значение критерия. Для удобства вычисления необходимо заполнить табл. 1.
4. Доказать, что
$$
\chi^2_{\text{набл}} = \sum \frac{n_i^2}{n_i'} - n.
$$
Проконтролировать корректность вычисления $\chi^2_{набл}$.
5. По заданному уровню значимости 
$\alpha = 0.05$ и числу степеней свободы $df$ найти критическую точку $\chi^2_{крит}$ и сравнить с наблюдаемым значением. Сделать выводы.


In [39]:
import numpy as np
import pandas as pd
import math
from scipy import stats

# Данные из ПР2
a = np.array([0.03, 0.053125, 0.07625, 0.099375, 0.1225, 0.145625, 0.16875, 0.191875])
b = np.array([0.053125, 0.07625, 0.099375, 0.1225, 0.145625, 0.16875, 0.191875, 0.215])
n_i = np.array([2, 2, 16, 12, 29, 19, 21, 7])
x_bar = 0.139201
s2 = 0.001433
s = 0.037851

N = int(n_i.sum())
x_i = (a + b) / 2

print(f"N = {N}")

N = 108


## Задача 1

Найдем точность и доверительный интервал для математического ожидания при неизвестном среднеквадратичном отклонении для доверительной точности $\gamma \in \{0.95, 0.99\}$.

При неизвестном СКО используем распределение Стьюдента:
$$
\bar x \pm t_{1-\alpha/2,\;n-1}\frac{s}{\sqrt n}.
$$

Найдём интервалы для $\gamma = 0.95$ и $\gamma = 0.99$.

In [40]:
gammas = [0.95, 0.99]
mu_rows = []

for gamma in gammas:
    alpha = 1 - gamma
    t_crit = stats.t.ppf(1 - alpha / 2, df=N - 1)
    eps = t_crit * s / math.sqrt(N)
    left = x_bar - eps
    right = x_bar + eps
    mu_rows.append([gamma, t_crit, eps, left, right])

mu_table = pd.DataFrame(mu_rows, columns=["γ", "t_кр", "ε", "Левая граница", "Правая граница"])
mu_table

,γ,t_кр,ε,Левая граница,Правая граница
0,0.95,1.982383,0.007220,0.131981,0.146421
1,0.99,2.622560,0.009552,0.129649,0.148753


С ростом доверительной вероятности интервал становится шире. Это видно при переходе от $\gamma = 0.95$ к $\gamma = 0.99$.

## Задача 2

Найдем значение $q$ при заданных $\gamma$ и $n$. Построим доверительные интервалы для среднеквадратичного отклонения.


Используем распределение $\chi^2$:
$$
\sqrt{\frac{(n-1)s^2}{\chi^2_{1-\alpha/2,\;n-1}}}
\le \sigma \le
\sqrt{\frac{(n-1)s^2}{\chi^2_{\alpha/2,\;n-1}}}.
$$

In [41]:
q_rows = []

for gamma in gammas:
    k = N - 1
    alpha = 1 - gamma
    chi2_left = chi2.ppf(alpha / 2, df=k)
    chi2_right = chi2.ppf(1 - alpha / 2, df=k)

    q = (np.sqrt((k) / chi2_left) - np.sqrt((k) / chi2_right)) / 2
    left = s * (1 - q)
    right = s * (1 + q)
    q_rows.append([gamma, chi2_left, chi2_right, q, left, right])

q_table = pd.DataFrame(q_rows, columns=["γ", "χ²_лево", "χ²_право", "q", "Левая граница СКО", "Правая граница СКО"])

q_table

,γ,χ²_лево,χ²_право,q,Левая граница СКО,Правая граница СКО
0,0.95,80.266785,137.516701,0.136244,0.032694,0.043008
1,0.99,73.074502,148.423613,0.180501,0.031019,0.044683


Построенные интервалы для среднеквадратичного отклонения при $\gamma = 0.95$ и $\gamma = 0.99$ не пересекаются, что говорит о том, что с ростом доверительной вероятности интервал становится шире.

Полученные значения для $q$ согласуются с табличными данными: полученные значения лежат между значениями для $n=100, 150$: $q(0.95, 100) = 0.143$, $q(0.99, 100) = 0.198$, $q(0.95, 150) = 0.115$, $q(0.99, 150) = 0.160$.

## Задача 3

Для крайних интервалов используем открытые границы:
- первый интервал: $(-\infty, b_1)$;
- последний интервал: $(a_k, +\infty)$.

Далее вычисляем:
$$
p_i = \Phi\left(\frac{b_i-\bar x}{s}\right) - \Phi\left(\frac{a_i-\bar x}{s}\right),
\qquad
n_i' = n p_i.
$$

In [42]:
p_i = []
interval_labels = []

for i in range(len(a)):
    left = -np.inf if i == 0 else a[i]
    right = np.inf if i == len(a) - 1 else b[i]

    p = stats.norm.cdf((right - x_bar) / s) - stats.norm.cdf((left - x_bar) / s)
    p_i.append(p)

    if i == 0:
        interval_labels.append(f"(-∞; {b[i]:.6f})")
    elif i == len(a) - 1:
        interval_labels.append(f"[{a[i]:.6f}; +∞)")
    else:
        interval_labels.append(f"[{a[i]:.6f}; {b[i]:.6f})")

p_i = np.array(p_i)
n_theor = N * p_i

pearson_table = pd.DataFrame({
    "i": np.arange(1, len(n_i) + 1),
    "Интервал": interval_labels,
    "n_i": n_i,
    "p_i": p_i,
    "n_i'": n_theor,
    "(n_i-n_i')^2": (n_i - n_theor) ** 2,
    "(n_i-n_i')^2/n_i'": (n_i - n_theor) ** 2 / n_theor,
    "n_i^2": n_i ** 2,
    "n_i^2/n_i'": n_i ** 2 / n_theor
})

pearson_table

,i,Интервал,n_i,p_i,n_i',(n_i-n_i')^2,(n_i-n_i')^2/n_i',n_i^2,n_i^2/n_i'
0,1,(-∞; 0.053125),2,0.011481,1.239921,0.577720,0.465933,4,3.226012
1,2,[0.053125; 0.076250),2,0.036663,3.959584,3.839968,0.969791,4,1.010207
2,3,[0.076250; 0.099375),16,0.098215,10.607255,29.081703,2.741680,256,24.134426
3,4,[0.099375; 0.122500),12,0.183164,19.781748,60.555605,3.061186,144,7.279438
4,5,[0.122500; 0.145625),29,0.237861,25.688973,10.962897,0.426755,841,32.737781
5,6,[0.145625; 0.168750),19,0.215117,23.232585,17.914772,0.771105,361,15.538521
6,7,[0.168750; 0.191875),21,0.135480,14.631819,40.553724,2.771612,441,30.139792
7,8,[0.191875; +∞),7,0.082020,8.858115,3.452592,0.389766,49,5.531651


In [43]:
sum_pearson_row = pd.DataFrame([{
    "i": "Σ",
    "Интервал": None,
    "n_i": n_i.sum(),
    "p_i": p_i.sum(),
    "n_i\'": n_theor.sum(),
    "(n_i-n_i')^2": None,
    "(n_i-n_i')^2/n_i'": pearson_table["(n_i-n_i')^2/n_i'"].sum(),
    "n_i^2": None,
    "n_i^2/n_i'": pearson_table["n_i^2/n_i'"].sum()
}])

pearson_table_pretty = pd.concat([pearson_table, sum_pearson_row], ignore_index=True)
pearson_table_pretty

,i,Интервал,n_i,p_i,n_i',(n_i-n_i')^2,(n_i-n_i')^2/n_i',n_i^2,n_i^2/n_i'
0,1,(-∞; 0.053125),2,0.011481,1.239921,0.57772,0.465933,4,3.226012
1,2,[0.053125; 0.076250),2,0.036663,3.959584,3.839968,0.969791,4,1.010207
2,3,[0.076250; 0.099375),16,0.098215,10.607255,29.081703,2.741680,256,24.134426
3,4,[0.099375; 0.122500),12,0.183164,19.781748,60.555605,3.061186,144,7.279438
4,5,[0.122500; 0.145625),29,0.237861,25.688973,10.962897,0.426755,841,32.737781
5,6,[0.145625; 0.168750),19,0.215117,23.232585,17.914772,0.771105,361,15.538521
6,7,[0.168750; 0.191875),21,0.135480,14.631819,40.553724,2.771612,441,30.139792
7,8,[0.191875; +∞),7,0.082020,8.858115,3.452592,0.389766,49,5.531651
8,Σ,None,108,1.000000,108.000000,None,11.597828,None,119.597828


## Задача 4

Основная формула:
$$
\chi^2_{набл} = \sum \frac{(n_i-n_i')^2}{n_i'}.
$$

Проверочная формула:
$$
\chi^2_{набл} = \sum \frac{n_i^2}{n_i'} - n.
$$

In [44]:
chi2_obs = pearson_table["(n_i-n_i')^2/n_i'"].sum()
chi2_obs_check = pearson_table["n_i^2/n_i'"].sum() - N

print(f"χ²_набл = {chi2_obs:.10f}")
print(f"Проверка по формуле Σ(n_i² / n_i') - n = {chi2_obs_check:.10f}")

χ²_набл = 11.5978284447
Проверка по формуле Σ(n_i² / n_i') - n = 11.5978284447


Оба способа дают одно и то же значение с точностью до округления. Значит, вычисления выполнены корректно.

## Задча 5

Уровень значимости:
$$
\alpha = 0.05.
$$

Число степеней свободы:
$$
df = k - 3.
$$

Критическое значение:
$$
\chi^2_{кр} = \chi^2_{0.95;4}.
$$

In [45]:
alpha = 0.05
k = len(n_i)

df = k - 3
chi2_crit = stats.chi2.ppf(1 - alpha, df=df)

print(f"df = {df}")
print(f"χ²_кр = {chi2_crit:.6f}")
print(f"χ²_набл: {chi2_obs:.6f}")

df = 5
χ²_кр = 11.070498
χ²_набл: 11.597828


Так как $\chi^2_{набл}$ > $\chi^2_{кр}$, гипотеза о нормальном распределении отвергается.

## Выводы

1. По сгруппированным данным найдены доверительные интервалы для математического ожидания при $\gamma = 0.95$ интервал получился $(0.132;\ 0.146)$ и $\gamma = 0.99$ - $(0.130;\ 0.149)$. При росте доверительной вероятности интервал становится шире.
2. Найдены доверительные интервалы для среднеквадратичного отклонения при $\gamma = 0.95$: $(0.032; 0.043)$ и $\gamma = 0.99$: $(0.031; 0.044)$. С ростом доверительной вероятности интервал также становится шире.
3. Для проверки гипотезы о нормальном распределении были вычислены теоретические частоты и значение статистики Пирсона.
4. После объединения первых двух интервалов получено:
$$
\chi^2_{\text{набл}} \approx 11.597, \qquad \chi^2_{\text{кр}} \approx 11.070.
$$
5. Так как $\chi^2_{\text{набл}} > \chi^2_{\text{кр}}$, гипотеза о нормальном распределении исследуемой случайной величины на уровне значимости $\alpha = 0.05$ отвергается.